tests/conftest.py

```diff
@@ -444,11 +445,17 @@ def __exit__(self, type, value, traceback):
             dag.__exit__(type, value, traceback)
             if type is None:
                 dag.clear()
+
                 self.dag_run = dag.create_dagrun(
                     run_id=self.kwargs.get("run_id", "test"),
                     state=self.kwargs.get('state', State.RUNNING),
                     execution_date=self.kwargs.get('execution_date', self.kwargs['start_date']),
                     start_date=self.kwargs['start_date'],
+                    run_type=self.kwargs.get('run_type', DagRunType.SCHEDULED),
+                    conf=self.kwargs.get('conf', {}),
+                    dag_hash=self.kwargs.get('dag_hash', None),
+                    external_trigger=self.kwargs.get('external_trigger', False),
+                    creating_job_id=self.kwargs.get('creating_job_id', None),
```

Comment:
I wonder if it’s better to store the kwargs explicitly instead. Either directly on `DagFactory`, or with a `NamedTuple` or `TypedDict`. This is getting unwieldy.

Reply:
It’s just all the arguments that dag.create_dagrun takes, I’m thinking since it’s just test we should leave it like this?

Reply:
Test code is still code 🙂 But I’m not too strong on this; we can always fix things later when they break.

Reply:
Cool. I'll try the NamedTuple when I rename the `dag_maker` to something like `dag_and_dagrun_creator` to be more reflective

Reply:
I think we should have two factories: `dag_and_dagrun_creator` (though I don't love the name) and a `dag_maker` -- `dag_maker` would be responsible for makeing the DAG, and ensuring that the SerliazedDag and DagModel tables are up-to-date, and then the `dag_and_dagrun_creator` can use that factory somehow, but also create a DagRun.